cmd/collector/app/span_processor_test.go

```diff
@@ -383,6 +383,142 @@ func TestSpanProcessorWithCollectorTags(t *testing.T) {
 	assert.Equal(t, expected.Process, span.Process)
 }
 
+func TestSpanProcessorWithCollectorTagsWithUseOnTagKeyConflictAsBoth(t *testing.T) {
+	testCollectorTags := map[string]string{
+		"extra": "tag",
+		"env":   "prod",
+		"node":  "172.22.18.161",
+	}
+
+	w := &fakeSpanWriter{}
+	p := NewSpanProcessor(w, nil, Options.CollectorTags(testCollectorTags), Options.UseOnTagKeyConflict("both")).(*spanProcessor)
+
+	defer assert.NoError(t, p.Close())
+	span := &model.Span{
+		Process: model.NewProcess("unit-test-service", []model.KeyValue{
+			{
+				Key:  "env",
+				VStr: "prod",
+			},
+			{
+				Key:  "node",
+				VStr: "k8s-test-node-01",
+			},
+		}),
+	}
+	p.addCollectorTags(span)
+	expected := &model.Span{
+		Process: model.NewProcess("unit-test-service", []model.KeyValue{
+			{
+				Key:  "env",
+				VStr: "prod",
+			},
+			{
+				Key:  "extra",
+				VStr: "tag",
+			},
+			{
+				Key:  "node",
+				VStr: "172.22.18.161",
+			},
+			{
+				Key:  "node",
+				VStr: "k8s-test-node-01",
+			},
+		}),
+	}
+
+	assert.Equal(t, expected.Process, span.Process)
+}
+
+func TestSpanProcessorWithCollectorTagsWithUseOnTagKeyConflictAsSpan(t *testing.T) {
+	testCollectorTags := map[string]string{
+		"extra": "tag",
+		"env":   "prod",
+		"node":  "172.22.18.161",
+	}
+
+	w := &fakeSpanWriter{}
+	p := NewSpanProcessor(w, nil, Options.CollectorTags(testCollectorTags), Options.UseOnTagKeyConflict("span")).(*spanProcessor)
+
+	defer assert.NoError(t, p.Close())
+	span := &model.Span{
+		Process: model.NewProcess("unit-test-service", []model.KeyValue{
```

Comment:
I think it would be nice to add an additional tag for each test's span that has a tag that doesn't exist in the `testCollectorTags`, which will catch cases in our logic that somehow assume overlapping keys or keys that only exist in the collector tags, i.e.:
```
{
	Key:  "only-in",
	VStr: "this-span",
},
```